base/pkg/dir.jl

```diff
@@ -28,7 +28,13 @@ function cd(f::Function, args...; kws...)
         !haskey(ENV,"JULIA_PKGDIR") ? init() :
             throw(PkgError("Package metadata directory $metadata_dir doesn't exist; run Pkg.init() to initialize it."))
     end
-    Base.cd(()->f(args...; kws...), dir)
+    if haskey(ENV,"JULIA_PKGDIR")
+        withenv("JULIA_PKGDIR" => _pkgroot()) do
```

Comment:
maybe this should just say `"JULIA_PKGDIR" => abspath(ENV["JULIA_PKGDIR"])` ? should be equivalent but takes less scrolling up to see what it's accomplishing